src/qinfer/hyper_heuristic_optimisers.py

```diff
@@ -0,0 +1,202 @@
+
+
+## FEATURES ###################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+## IMPORTS ####################################################################
+
+import numpy as np
+import random
+from functools import partial
+from qinfer.perf_testing import perf_test_multiple
+from qinfer import distributions
+    
+## CLASSES ####################################################################
+
+__all__ = [
+    'ParticleSwarmOptimiser'
+]
+
+class HyperHeuristicOptimiser(object):
+    '''
+        A generic hyper-heuristic optimiser class that is inherited by the other optimisation functions.
+
+        :param np.ndarray param_names: The list of parameters that are being searched over.
+        :param function fitness_function: The function that is being optimised over, defaults to perf test multiple
+        :param function boundary_map: Function to constrain points within some boundary regime
+        :param dict funct_args: Arguments to pass to the fitness function
+        :param dict funct_kwargs: Keyword arguments to pass to the fitness function
+    '''
+
+    def __init__(
+                self,
+                param_names,
+                fitness_function = None,
+                boundary_map=None,
+                *funct_args,
+                **funct_kwargs
+                ):
+        self._param_names = param_names
+        self._n_free_params = len(param_names)
+        self._boundary_map = boundary_map
+        self._funct_args = funct_args
+        self._funct_kwargs = funct_kwargs
+        
+        if fitness_function is None: # Default to calling perf test multiple
+            self._optimisable = PerfTestMultipleAbstractor(
+                self._param_names,
+                *self._funct_args,
+                **self._funct_kwargs
+            )
+        else: 
+            self._fitness_function = partial(fitness_function, *self._funct_args, **self._funct_kwargs)
+
+    # Member function needed for parralelisation
+    def fitness_function(self, params):
+        return self._fitness_function(params)
+    
+    def parrallel(self):
+        raise NotImplementedError("This optimiser does not have parrallel support. To resolve this issue, level an appropriate criticism at the developer.")
+
+class ParticleSwarmOptimiser(HyperHeuristicOptimiser):
+    '''
+        A particle swarm optimisation based hyperheuristic
```

Comment:
Good start to the docstring, but we should make sure it describes all of the kwargs.